java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -0,0 +1,656 @@
+        }
+        // process body
+        currentBlock = createBlock(currentBlock);
+        continueTargets.addLast(stepBlock);
+        breakTargets.addLast(falseBranch);
+        build(s.statement());
+        breakTargets.removeLast();
+        continueTargets.removeLast();
+        Block body = currentBlock;
+        // process condition
+        ExpressionTree condition = s.condition();
+        if (condition != null) {
+          currentBlock = createBranch(s, body, falseBranch);
+          buildCondition(condition, body, falseBranch);
+        } else {
+          currentBlock = createUnconditionalJump(s, body);
+        }
+        stepBlock.successors.add(currentBlock);
+        // process init
+        currentBlock = createBlock(currentBlock);
+        for (StatementTree init : Lists.reverse(s.initializer())) {
+          build(init);
+        }
+        break;
+      }
+      case TRY_STATEMENT: {
+        //FIXME only path with no failure constructed for now, (not taking try with resources into consideration).
+        TryStatementTree tryStatementTree = (TryStatementTree) tree;
+        currentBlock = createBlock(currentBlock);
+        BlockTree finallyBlock = tryStatementTree.finallyBlock();
+        if(finallyBlock != null) {
+          build(finallyBlock);
+        }
+        currentBlock = createBlock(currentBlock);
+        build(tryStatementTree.block());
+        break;
+      }
+      case THROW_STATEMENT: {
+        //FIXME this won't work if it is intended to be caught by a try statement.
+        ThrowStatementTree throwStatementTree = (ThrowStatementTree) tree;
+        currentBlock = createUnconditionalJump(throwStatementTree, exitBlock);
+        build(throwStatementTree.expression());
+        break;
+      }
+      case SYNCHRONIZED_STATEMENT: {
+        SynchronizedStatementTree sst = (SynchronizedStatementTree) tree;
+        //Naively build synchronized statement.
+        build(sst.block());
+        build(sst.expression());
+        break;
+      }
+      case POSTFIX_INCREMENT:
+      case POSTFIX_DECREMENT:
+      case PREFIX_INCREMENT:
+      case PREFIX_DECREMENT:
+      case UNARY_MINUS:
+      case UNARY_PLUS:
+      case BITWISE_COMPLEMENT:
+      case LOGICAL_COMPLEMENT:
+        UnaryExpressionTree e = (UnaryExpressionTree) tree;
+        currentBlock.elements.add(e);
+        build(e.expression());
+        break;
+      case PARENTHESIZED_EXPRESSION:
+        build(((ParenthesizedTree) tree).expression());
+        break;
+      case ARRAY_ACCESS_EXPRESSION: {
+        ArrayAccessExpressionTree aaet = (ArrayAccessExpressionTree) tree;
+        currentBlock.elements.add(aaet);
+        build(aaet.expression());
+        build(aaet.dimension());
+        break;
+      }
+      case ARRAY_DIMENSION: {
+        ArrayDimensionTree arrayDimensionTree = (ArrayDimensionTree) tree;
+        if(arrayDimensionTree.expression() != null) {
+          build(arrayDimensionTree.expression());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 13. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)